### PR TITLE
Replace `println!` from library functions with configurable logging

### PR DESCRIPTION
**Pull request by mishazharov**
Developers building on this crate do not necessarily need logging to occur over `println!` when authenticating 2FA, so this commit replaces the print macros with logging macros

Fixes #

- [x] cargo test has been run and passes
- [x] documentation has been updated with relevant examples (if relevant)

### CODE REVIEW

File: webauthn-authenticator-rs/src/mozilla.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ impl MozillaAuthenticator {
                         continue;
                     }
                     PinError::InvalidPin(attempts) => {
-                        println!(
+                        debug!(
```
Comment (yaleman) [inline webauthn-authenticator-rs/src/mozilla.rs:83]: this should be an error

**Review by yaleman — CHANGES_REQUESTED** [review]

File: webauthn-authenticator-rs/src/mozilla.rs (hunk 1 above)
Comment (mishazharov) [reply webauthn-authenticator-rs/src/mozilla.rs]: Thanks for taking a look, updated!

**Comment by mishazharov** [comment]
It looks like there's some CI failures unrelated to this change:
```
error: package `bumpalo v3.15.4` cannot be built because it requires rustc 1.73.0 or newer, while the currently active rustc version is 1.70.0
Either upgrade to rustc 1.73.0 or newer, or use
cargo update -p bumpalo@3.15.4 --precise ver
where `ver` is the latest version of `bumpalo` supporting rustc 1.70.0
Error: Process completed with exit code 101.
```

**Comment by yaleman** [comment]
@Firstyear how's this one look? Figured I'd ask because you've used it more

**Review by yaleman — DISMISSED** [review]

File: webauthn-authenticator-rs/src/mozilla.rs [hunk 2]
```diff
@@ -58,19 +58,19 @@ impl MozillaAuthenticator {
...
-                    println!("STATUS: device unavailable: {}", dev_info)
+                    debug!("STATUS: device unavailable: {}", dev_info)
                 }
                 Ok(StatusUpdate::Success { dev_info }) => {
-                    println!("STATUS: success using device: {}", dev_info);
+                    debug!("STATUS: success using device: {}", dev_info);
                 }
                 Ok(StatusUpdate::SelectDeviceNotice) => {
-                    println!("STATUS: Please select a device by touching one of them.");
+                    debug!("STATUS: Please select a device by touching one of them.");
```
Comment (Firstyear) [inline webauthn-authenticator-rs/src/mozilla.rs:70]: I think this one needs to be info! at the least.

**Review by Firstyear — COMMENTED** [review]
Actually @micolous would be the person to ask. There are some places where I think we use these messages as communication points for cli tools.

**Comment by micolous** [comment]
> Actually @micolous would be the person to ask. There are some places where I think we use these messages as communication points for cli tools.

All the other backends use `info!`, `error!`, etc.; the Mozilla backend is the only one which *doesn't*.

The CLI tools I wrote touch the other backends (as they need to interact with it on a CTAP2 level, rather than an "Authenticator" level), so this would at least make things consistent (as they will eventually replace the Mozilla backend).

I believe the main user of the Mozilla backend is Kanidm's CLI tools – I don't know what makes sense there.

**Comment by yaleman** [comment]
Libraries really should use the `log` primitives instead of `println` so the dependent apps can do what they need to from there...

**Review by Firstyear — APPROVED** [review]

**Pull request merged**